spec/system/nested_documentable_spec.rb

```diff
@@ -0,0 +1,314 @@
+        fill_in "Title", with: "My Title"
+        attach_file "Choose document", file_fixture("empty.pdf")
+
+        expect(page).to have_css ".loading-bar.complete"
+      end
+
+      expect(page).to have_field("Title", with: "My Title")
+    end
+
+    scenario "Should update document cached_attachment field after valid file upload" do
+      do_login_for(user, management: management_section?(path))
+      visit path
+
+      click_link "Add new document"
+
+      cached_attachment_field = find("input[name$='[cached_attachment]']", visible: :hidden)
+      expect(cached_attachment_field.value).to be_empty
+
+      attach_file "Choose document", file_fixture("empty.pdf")
+
+      expect(page).to have_css(".loading-bar.complete")
+      expect(cached_attachment_field.value).not_to be_empty
+    end
+
+    scenario "Should not update document cached_attachment field after invalid file upload" do
+      do_login_for(user, management: management_section?(path))
+      visit path
+
+      documentable_attach_new_file(file_fixture("logo_header.gif"), success: false)
+
+      cached_attachment_field = find("input[name$='[cached_attachment]']", visible: :hidden)
+      expect(cached_attachment_field.value).to be_empty
+    end
+
+    scenario "Should show document errors after documentable submit with empty document fields" do
+      do_login_for(user, management: management_section?(path))
+      visit path
+
+      click_link "Add new document"
+      click_button submit_button_text
+
+      within "#nested-documents .document-fields" do
+        expect(page).to have_content("can't be blank", count: 2)
+      end
+    end
+
+    scenario "Should delete document after valid file upload and click on remove button" do
+      do_login_for(user, management: management_section?(path))
+      visit path
+
+      documentable_attach_new_file(file_fixture("empty.pdf"))
+      click_link "Remove document"
+
+      expect(page).not_to have_css("#nested-documents .document-fields")
+    end
+
+    scenario "Should show successful notice when resource filled correctly without any nested documents" do
+      do_login_for(user, management: management_section?(path))
+      visit path
+
+      fill_in_required_fields(factory, path)
+      click_button submit_button_text
+
+      expect(page).to have_content notice_text
+    end
+
+    scenario "Should show successful notice when resource filled correctly and after valid file uploads" do
+      Setting["uploads.documents.max_amount"] = 3
+      do_login_for(user, management: management_section?(path))
+      visit path
+
+      fill_in_required_fields(factory, path)
+
+      %w[clippy empty logo].take(documentable.class.max_documents_allowed).each do |filename|
+        click_link "Add new document"
+
+        attach_file "Choose document", file_fixture("#{filename}.pdf")
+
+        within all(".document-fields").last do
+          expect(page).to have_css ".loading-bar.complete"
+        end
+      end
```

Comment:
Maybe we could add an extra expectation here to check that there's no longer a button to add a document :thinking:. Since this is the only test with multiple uploads...